pkgs/common-updater/scripts/list-archive-versions

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env bash
+
+# lists all available versions listed for a package in a site (http)
+
+archive=""  # archive url
+pname=""  # package name
+file=""  # file for writing debugging information
+
+while (( $# > 0 )); do
+    flag="$1"
+    shift 1
+    case "$flag" in
+        --url=*)
+            archive="${flag#*=}"
+            ;;
+        --pname=*)
+            pname="${flag#*=}"
+            ;;
+        --file=*)
+            file="${flag#*=}"
+            ;;
+        *)
+            echo "$0: unknown option ‘${flag}’"
+            exit 1
+            ;;
+    esac
+done
+
+# by default set url to the base dir of the first url in src.urls
+if [[ -z "$archive" ]]; then
+    archive="$(nix-instantiate $systemArg --eval -E \
+                   "with import ./. {}; dirOf (dirOf (lib.head $UPDATE_NIX_ATTR_PATH.src.urls))" \
+            | tr -d '"')"
+fi
+
+if [[ -z "$pname" ]]; then
+    pname="$UPDATE_NIX_ATTR_PATH"
+fi
+
+set +x
+
+# print a debugging message
+if [[ -n "$file" ]]; then
+    echo "# Listing versions for '$pname' at $archive" >> $file
```

Comment:
It also prints to this file `git add` and `git commit` commands.

Normally I update packages with a command like the following:
``` console
$ nix-shell maintainers/scripts/update.nix --argstr path xfce
```
which prints to the file `update-git-commits.txt`. General information (as the above), as well as git add and commit commands are print in it. Usually I source this file in order to commit the changes if I think the updates are all right.

Because of that I would keep printing like this. Unless there is another good way of getting the git commands.

Reply:
I usually just run `nix-shell maintainers/scripts/update.nix --arg predicate '(path: pkg: pkg.updateScript.name or null == "gnome-update-script")' --argstr keep-going true --argstr commit true` and then drop the irrelevant commits in interactive rebase.